File: ChipTallyiOS/ChipTally/Domain/UseCases/CreateGameSessionUseCase.swift
//
//  CreateGameSessionUseCase.swift
//  ChipTally
//

import Foundation

protocol CreateGameSessionUseCaseProtocol {
    func execute(playerNames: [String], initialChipCount: Int) -> GameSession
}

final class CreateGameSessionUseCase: CreateGameSessionUseCaseProtocol {
    func execute(playerNames: [String], initialChipCount: Int) -> GameSession {
        let players = playerNames.map { name in
            Player(name: name, chipCount: initialChipCount)
        }
        return GameSession(players: players, initialChipCount: initialChipCount)
    }
}

File: ChipTallyAOS/app/src/main/java/com/chiptally/domain/usecase/CreateGameSessionUseCase.kt
package com.chiptally.domain.usecase

import com.chiptally.domain.model.GameSession
import com.chiptally.domain.model.Player

class CreateGameSessionUseCase {
    fun execute(playerNames: List<String>, initialChipCount: Int): GameSession {
        val players = playerNames.map { name ->
            Player(name = name, chipCount = initialChipCount)
        }.toMutableList()
        return GameSession(players = players, initialChipCount = initialChipCount)
    }
}
